pkg/breakglasscredential/flags.go

```diff
@@ -0,0 +1,153 @@
+package breakglasscredential
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+const (
+	usernameFlag   = "username"
+	expirationFlag = "expiration"
+)
+
+type BreakGlassCredentialArgs struct {
+	username           string
+	expirationDuration time.Duration
+}
+
+func AddBreakGlassCredentialFlags(cmd *cobra.Command) *BreakGlassCredentialArgs {
+	args := &BreakGlassCredentialArgs{}
+
+	cmd.Flags().StringVar(
+		&args.username,
+		usernameFlag,
+		"",
+		"Username for the break glass credential.",
+	)
+
+	cmd.Flags().DurationVar(
+		&args.expirationDuration,
+		expirationFlag,
+		0,
+		fmt.Sprintf("Expire the break glass credential after a relative duration like 2h, 8h. "+
+			"The expiration duration needs to be at least 10 minutes from now and to be at maximum 24 hours."),
+	)
+	return args
+}
+
+func GetBreakGlassCredentialOptions(cmd *pflag.FlagSet, args *BreakGlassCredentialArgs) (
+	*BreakGlassCredentialArgs, error) {
+	var err error
+	result := &BreakGlassCredentialArgs{}
+
+	result.username = args.username
+	result.expirationDuration = args.expirationDuration
+
+	if !IsBreakGlassCredentialSetViaCLI(cmd) {
+		if !interactive.Enabled() {
+			return nil, nil
+		}
+	}
+
+	if interactive.Enabled() && !cmd.Changed(usernameFlag) {
+		result.username, err = interactive.GetString(interactive.Input{
+			Question: "Username",
+			Default:  result.username,
+			Help:     cmd.Lookup(usernameFlag).Usage,
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if interactive.Enabled() && !cmd.Changed(expirationFlag) {
+		inputString, err := interactive.GetString(interactive.Input{
+			Question: "Expiration duration",
+			Default:  result.expirationDuration,
+			Help:     cmd.Lookup(expirationFlag).Usage,
+		})
+		if err != nil {
+			return nil, err
+		}
+		if inputString != "" {
+			duration, err := time.ParseDuration(inputString)
+			if err != nil {
+				return nil, err
+			}
+			result.expirationDuration = duration
+		}
+	}
+
+	return result, nil
+}
+
+func CreateBreakGlass(cluster *cmv1.Cluster,
+	clusterKey string,
+	args *BreakGlassCredentialArgs, r *rosa.Runtime) error {
+
+	breakGlassConfig, err := CreateBreakGlassConfig(args)
+	if err != nil {
+		return fmt.Errorf("failed to create a break glass credential for cluster '%s': %s",
+			clusterKey, err)
+	}
+
+	_, err = r.OCMClient.CreateBreakGlassCredential(cluster.ID(), breakGlassConfig)
+
+	if err != nil {
+		return fmt.Errorf("failed to create a break glass credential for cluster '%s': %s",
+			clusterKey, err)
+	}
+	return nil
+
+}
+
+func CreateBreakGlassConfig(args *BreakGlassCredentialArgs) (*cmv1.BreakGlassCredential, error) {
```

Comment:
@chenz4027 It would be good to get some test coverage on these functions 👍

Reply:
Added the test coverage